sdk/azcore/internal/shared/expiring_resource.go

```diff
@@ -28,6 +28,9 @@ type ExpiringResource struct {
 	// expiration indicates when the shared resource expires; it is 0 if the resource was never acquired
 	expiration time.Time
 
+	// lastAcquired indicates when a thread/goroutine last attempted to acquire/update the resource
+	lastAcquired time.Time
```

Comment:
Nit, from the usage and comment I think this would be better named `lastAttempt`